parseAPI/src/IA_amdgpu.C

```diff
@@ -0,0 +1,150 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "IA_amdgpu.h"
+#include "Register.h"
+#include "Dereference.h"
+#include "Immediate.h"
+#include "BinaryFunction.h"
+
+#include "common/src/arch.h"
+
+#include "parseAPI/src/debug_parse.h"
+
+#include <deque>
+#include <iostream>
+#include <sstream>
+#include <functional>
+#include <algorithm>
+#include <set>
+
+using namespace Dyninst;
+using namespace InstructionAPI;
+using namespace Dyninst::ParseAPI;
+using namespace Dyninst::InsnAdapter;
+
+IA_amdgpu::IA_amdgpu(Dyninst::InstructionAPI::InstructionDecoder dec_,
+               Address start_, 
+	       Dyninst::ParseAPI::CodeObject* o,
+	       Dyninst::ParseAPI::CodeRegion* r,
+	       Dyninst::InstructionSource *isrc,
+	       Dyninst::ParseAPI::Block * curBlk_):
+	           IA_IAPI(dec_, start_, o, r, isrc, curBlk_) {
+}		   
+IA_amdgpu::IA_amdgpu(const IA_amdgpu& rhs): IA_IAPI(rhs) {}
+
+IA_amdgpu* IA_amdgpu::clone() const {
+    return new IA_amdgpu(*this);
+}
+
+bool IA_amdgpu::isFrameSetupInsn(Instruction i) const
+{
+    return false;
+}
+
+
+bool IA_amdgpu::isNop() const
+{
+    Instruction ci = curInsn();
+    if(ci.getOperation().getID() == amdgpu_op_s_nop)
+        return true;
+    return false;
+}
+
+bool IA_amdgpu::isThunk() const 
+{
+    return false;
+}
+
+bool IA_amdgpu::isTailCall(const Function* context, EdgeTypeEnum type, unsigned int,
+        const std::set<Address>& knownTargets ) const
+{
+   return false;
+}
+
+bool IA_amdgpu::savesFP() const
+{
+    return false;
+}
+
+bool IA_amdgpu::isStackFramePreamble() const
+{
+    return false;
+}
+
+bool IA_amdgpu::cleansStack() const
+{
+   return false;
+}
+
+bool IA_amdgpu::sliceReturn(ParseAPI::Block* bit, Address ret_addr, ParseAPI::Function * func) const
+{
+    return true;
+}
+
+bool IA_amdgpu::isReturnAddrSave(Address& retAddr) const
+{
+  return false;
+}
+
+bool IA_amdgpu::isReturn(Dyninst::ParseAPI::Function * context, Dyninst::ParseAPI::Block* currBlk) const
+{
+    return curInsn().getCategory() == c_ReturnInsn;
+}
+
+bool IA_amdgpu::isFakeCall() const
+{
+    return false;
+}
+
+bool IA_amdgpu::isIATcall(std::string &) const
+{
+    return false;
+}
+
+bool IA_amdgpu::isLinkerStub() const
+{
+  // Disabling this code because it ends with an
+    // incorrect CFG.
+    return false;
+}
+
+#if 0
```

Comment:
If this code isn't needed, then let's remove it.

Reply:
removed